python/lsst/obs/base/yamlCamera.py

```diff
@@ -23,181 +23,318 @@
 
 import numpy as np
 import lsst.afw.cameraGeom as cameraGeom
+import lsst.geom as geom
 import lsst.afw.geom as afwGeom
 from lsst.afw.table import AmpInfoCatalog, AmpInfoTable
-from lsst.afw.cameraGeom.cameraFactory import makeDetector
+from lsst.afw.cameraGeom.cameraFactory import makeDetectorData
 
+__all__ = ["makeCamera"]
 
-class YamlCamera(cameraGeom.Camera):
-    """The Commissioning Camera (comCam)
+
+def makeCamera(cameraFile):
+    """An imaging camera (e.g. the LSST 3Gpix camera)
 
     Parameters
     ----------
-    cameraYamlFile : `str`
+    cameraFile : `str`
         Camera description YAML file.
+
+    Return a `lsst.afw.cameraGeom.Camera`
+    """
+
+    with open(cameraFile) as fd:
+        cameraParams = yaml.load(fd, Loader=yaml.Loader)
+
+    cameraName = cameraParams["name"]
+
+    #
+    # Handle distortion models.
+    #
+    plateScale = afwGeom.Angle(cameraParams["plateScale"], afwGeom.arcseconds)
+    nativeSys = cameraGeom.CameraSys(cameraParams["transforms"].pop("nativeSys"))
+    transforms = makeTransformDict(nativeSys, cameraParams["transforms"], plateScale)
+
+    ccdParams = cameraParams["CCDs"]
+    detectorConfigList = makeDetectorConfigList(ccdParams)
+
+    ampInfoCatDict = {}
+    for (ccdName, ccdValues), detectorConfig in zip(ccdParams.items(), detectorConfigList):
```

Comment:
I may be missing something obvious but `detectorConfig` isn't used in the loop. Also, if it was being used it requires that the items are read out the dict in the order expected in `detectorConfigList`. Python 3.7 declares that dicts are now insertion ordered and 3.6 behaves like that but unofficially, but this loop does not seem right to me.